### PR TITLE
Add WithRandomColor to EmbedBuilderExtensions

### PR DESCRIPTION
**Pull request by Tesseract-arch**

### CODE REVIEW

File: Discord.Net.Extensions/Core/Embeds/EmbedBuilderExtensions.cs [hunk 1]
```diff
@@ -26,8 +26,10 @@ MIT License
 
 namespace Discord.Extensions
 {
-    internal static class EmbedBuilderExtensions
+    public static class EmbedBuilderExtensions
     {
-        // TODO
+        private static readonly Random random = new();
```
Comment (Rozen4334) [inline Discord.Net.Extensions/Core/Embeds/EmbedBuilderExtensions.cs:31]:
```suggestion
        private static readonly Lazy<Random> _random = new(() => new());
```

File: Discord.Net.Extensions/Core/Embeds/EmbedBuilderExtensions.cs [hunk 2]
```diff
@@ -26,8 +26,10 @@ MIT License
...
+
+        public static EmbedBuilder WithRandomColor(this EmbedBuilder builder) => builder.WithColor(new Color((uint)random.Next(0xFFFFFF)));
```
Comment (Rozen4334) [inline Discord.Net.Extensions/Core/Embeds/EmbedBuilderExtensions.cs:33]:
```suggestion
        public static EmbedBuilder WithRandomColor(this EmbedBuilder builder) => builder.WithColor(new Color((uint)_random.Value.Next(0xFFFFFF)));
```

**Review by Rozen4334 — CHANGES_REQUESTED** [review]
Consider making the Random lazy so it doesn't get initialized until called.

**Comment by Tesseract-arch** [comment]
> Consider making the Random lazy so it doesn't get initialized until called.

Good idea, updated.

**Pull request merged**